src/System.Drawing.Primitives/src/System/Drawing/Color.cs

```diff
@@ -420,188 +420,119 @@ private long Value
+                ThrowOutOfByteRange(value, name);
         }
 
-        private static long MakeArgb(byte alpha, byte red, byte green, byte blue) =>
-            (long)unchecked((uint)(red << ARGBRedShift |
+        private static uint MakeArgb(byte alpha, byte red, byte green, byte blue) =>
+            unchecked((uint)(red << ARGBRedShift |
                 green << ARGBGreenShift |
                 blue << ARGBBlueShift |
-                alpha << ARGBAlphaShift)) & 0xffffffff;
+                alpha << ARGBAlphaShift));
 
-        public static Color FromArgb(int argb) => new Color(argb & 0xffffffff, StateARGBValueValid, null, 0);
+        private static Color FromArgb(uint argb) => new Color(argb, StateARGBValueValid, null, (KnownColor)0);
+
+        public static Color FromArgb(int argb) => FromArgb((uint)argb);
 
         public static Color FromArgb(int alpha, int red, int green, int blue)
         {
             CheckByte(alpha, nameof(alpha));
             CheckByte(red, nameof(red));
             CheckByte(green, nameof(green));
             CheckByte(blue, nameof(blue));
-            return new Color(MakeArgb((byte)alpha, (byte)red, (byte)green, (byte)blue), StateARGBValueValid, null, (KnownColor)0);
+
+            return FromArgb(MakeArgb((byte)alpha, (byte)red, (byte)green, (byte)blue));
         }
 
         public static Color FromArgb(int alpha, Color baseColor)
         {
             CheckByte(alpha, nameof(alpha));
-            // unchecked - because we already checked that alpha is a byte in CheckByte above
-            return new Color(MakeArgb(unchecked((byte)alpha), baseColor.R, baseColor.G, baseColor.B), StateARGBValueValid, null, (KnownColor)0);
+
+            return FromArgb(unchecked((uint)alpha << ARGBAlphaShift | ((uint)baseColor.Value & ~ARGBAlphaMask)));
         }
 
-        public static Color FromArgb(int red, int green, int blue) => FromArgb(255, red, green, blue);
+        public static Color FromArgb(int red, int green, int blue) => FromArgb(byte.MaxValue, red, green, blue);
 
         public static Color FromKnownColor(KnownColor color) =>
             color <= 0 || color > KnownColor.MenuHighlight ? FromName(color.ToString()) : new Color(color);
 
         public static Color FromName(string name)
         {
             // try to get a known color first
-            Color color;
-            if (ColorTable.TryGetNamedColor(name, out color))
-            {
+            if (ColorTable.TryGetNamedColor(name, out Color color))
                 return color;
-            }
+
             // otherwise treat it as a named color
             return new Color(NotDefinedValue, StateNameValid, name, (KnownColor)0);
         }
 
         public float GetBrightness()
         {
-            float r = R / 255.0f;
-            float g = G / 255.0f;
-            float b = B / 255.0f;
+            uint value = (uint)Value;
+            int r = (int)((value >> ARGBRedShift  ) & 0xff);
```

Comment:
You could just do `(int)R`, `(int)G`, `(int)B` or why not use `byte Math.Min(byte, byte);`

Reply:
> You could just do (int)R, (int)G, (int)B

The R, G, and B accessors each make a separate call to the fairly costly Value accessor.  I retrieve Value only once.

> why not use byte Math.Min(byte, byte);

The values will be cast to `int` later for addition or subtraction anyway.  May as well start out as `int`

Reply:
> The R, G, and B accessors each make a separate call to the fairly costly Value accessor. I retrieve Value only once.

That is true. 

Would it make sense to create a private method for this? This code is used in GetBrightness, GetHue and GetSaturation? something like `(int r, int g, int b) GetRGBFromValue()` ? Or that would hurt the perf gain?

Reply:
I thought about that, but it seemed a bit clumsy with the triple return.  Using the existing property accessors, I noticed `B` was inlined while `R` and `G` were not.  I would assume an all-in-one helper would also not be inlined, so that would eat up at least part of the perf gain.  I can try it out and see how much.

Reply:
Tried the `ValueTuple` approach, but it ate up most of the perf gains.  I was able to keep them with forced inline helper using `out` params, though.  Bit cleaner that way...

Reply:
Yes it indeed looks cleaner.